### PR TITLE
[ml-release][no_ci] fix a few more ml user test.

### PR DESCRIPTION
**Pull request by xwjiang2010**
Adjust horovod test and tune-rllib test instance number.

Signed-off-by: xwjiang2010 <xwjiang2010@gmail.com>

## Why are these changes needed?

Adjust horovod test and tune-rllib test instance number.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/ml_user_tests/horovod/compute_tpl.yaml [hunk 1]
```diff
@@ -5,13 +5,13 @@ max_workers: 3
 
 head_node_type:
     name: head_node
-    instance_type: g3.8xlarge
+    instance_type: m5.4xlarge
```
Comment (xwjiang2010) [inline release/ml_user_tests/horovod/compute_tpl.yaml:8]: @amogkam it seems we don't need a GPU head node unless I am miscalculating.

We are running 6 workers. g3.8xlarge has 2 GPUs per node. So 3 workers alone should cut it?

**Comment by xwjiang2010** [comment]
@gjoliver could you sign off the part of removing tune_rllib? Thanks!

**Review by gjoliver — APPROVED** [review]
for tune_rllib

**Comment by amogkam** [comment]
@xwjiang2010 can we run the release tests before merging?

**Comment by xwjiang2010** [comment]
tests running at : https://buildkite.com/ray-project/release-tests-pr/builds/21495#01848322-b9b9-4b6b-865c-333b43ebf370

**Pull request merged**